File: customwidget/src/main/java/cn/beauty/customwidget/MainActivity.java
package cn.beauty.customwidget;

import android.graphics.Canvas;
import android.graphics.Matrix;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Matrix matrix = new Matrix();
        matrix.postTranslate(100f, 100f);

        float src[] = {50f, 50f};
        float[] dst = new float[2];
        matrix.mapPoints(dst, src);

        Canvas canvas = new Canvas();
        Log.i("dstX", dst[0] + "");
        Log.i("dstY", dst[1] + "");
    }
}

File: architecture/src/main/java/com/example/meng/architecture/MainActivity.kt
package com.example.meng.architecture

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProvider
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.PersistableBundle
import com.example.meng.architecture.mvvm.MyViewModel
import com.example.meng.architecture.mvvm.User

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // update UI
        val model = ViewModelProvider.AndroidViewModelFactory.getInstance(this.application).create(MyViewModel::class.java)
        model.getUsers().observe(this, Observer<List<User>> {


        })
    }

    override fun onSaveInstanceState(outState: Bundle?, outPersistentState: PersistableBundle?) {
        super.onSaveInstanceState(outState, outPersistentState)
    }


}
